### ISSUE TITLE
[BUG] Allow `disable-line` to suppress violations caused by itself

### ISSUE THREAD

**Issue by webknjaz** [adrienverge/yamllint#745]
*This is similar to #343 (at least it seems so, I'll let you judge that).*

I have a commented out mapping key (`include:`) that follows an uncommented key (`exclude:`) that has a sequence as a value (specifically `# ` prepended to https://github.com/cherrypy/cherrypy/blob/88987cb/.github/workflows/ci-cd.yml#L604C9-L604C59; it's not yet commented out in the linked commit but is going to be because it makes `check-github-workflows` in https://github.com/python-jsonschema/check-jsonschema unhappy, but that's another story — I'm only mentioning this for context).

I obviously don't want it indented because it'd look like it's a part of the sequence of the previous key, which it isn't.

To facilitate this, I was trying to disable the rule on the line via `# yamllint disable-line rule:comments-indentation` and exceeded the line length. So I tried adding that rule to ignore as well.

My problem is that it didn't work.

Here's the repro:
```yaml
# repro.yml
---
top:
  matrix:
  - some: val
    key: another
  # include: []  # TODO: uncomment blah blah blaaaaaaaaaaah  # yamllint disable-line rule:comments-indentation rule:line-length
```

```console
$ yamllint repro.yml
repro.yml
  6:3       warning  comment not indented like content  (comments-indentation)
  6:81      error    line too long (127 > 80 characters)  (line-length)
```

So it sounds like there's two issues with this:
1) disabling rules doesn't seem to apply to comment lines
2) disabling rules generates more violations

I think that both of these should be thought though: should yamllint auto-suppress line length violations when they are called by `# yamllint` comments? Should it at least apply such suppressions to the comment portions of lines?

**Comment by webknjaz** [comment]
*(a workaround of the disable-violate-enable sandwich works, but feels clumsy)*

**Comment by adrienverge** [comment]
Hello Sviatoslav,

I believe this problem appears because yamllint disable directives *must start the comment*, which means it's pretty hard to apply them on YAML comments themselves. In other words, `# TODO: blah blah  # yamllint disable-line` doesn't work, because the comment doesn't start exactly with `# yamllint disable-line`.
In my opinion, it would make sense to allow such directives *inside* comments,
as long as the follow a `# `.

To answer your questions precisely:
1. True. This behavior was intended, but I feel it makes sense to change it.
2. I don't think that yamllint should decide to auto-suppress line length violations by itself.

   Moreover, there's an easy way to suppress such line length overruns. When line length becomes a problem, you can place the directive `# yamllint disable-line` *above* the target line, like this:
   ```yaml
   # yamllint disable-line
   line: to ignore
   ```

**Comment by webknjaz** [comment]
> Moreover, there's an easy way to suppress such line length overruns. When line length becomes a problem, you can place the directive # yamllint disable-line above the target line, like this:\n\n# yamllint disable-line\nline: to ignore

I forgot to mention that doing so in the above scenario introduced another indent violation when I tried it, which is why I've attempted it inline.
